test/src/main/java/org/apache/accumulo/test/functional/BulkFailureIT.java

```diff
@@ -0,0 +1,265 @@
+    String testFile = createTestFile(testData, fs);
+
+    c.tableOperations().create(table);
+    String tableId = c.tableOperations().tableIdMap().get(table);
+
+    // Table has no splits, so this extent corresponds to the tables single tablet
+    KeyExtent extent = new KeyExtent(tableId.toString(), null, null);
+
+    // Set up site configuration because this test uses server side code that expects it.
+    setupSiteConfig();
+
+    long fateTxid = 99999999L;
+
+    AccumuloServerContext asCtx = new AccumuloServerContext(
+        new ServerConfigurationFactory(HdfsZooInstance.getInstance()));
+    ZooArbitrator.start(Constants.BULK_ARBITRATOR_TYPE, fateTxid);
+
+    VolumeManager vm = VolumeManagerImpl.get();
+
+    // move the file into a directory for the table and rename the file to something unique
+    String bulkDir = BulkImport.prepareBulkImport(asCtx, vm, testFile, tableId);
+
+    // determine the files new name and path
+    FileStatus status = fs.listStatus(new Path(bulkDir))[0];
+    Path bulkLoadPath = fs.makeQualified(status.getPath());
+
+    // Directly ask the tablet to load the file.
+    assignMapFiles(fateTxid, asCtx, extent, bulkLoadPath.toString(), status.getLen());
+
+    assertEquals(ImmutableSet.of(bulkLoadPath), getFiles(c, extent));
+    assertEquals(ImmutableSet.of(bulkLoadPath), getLoaded(c, extent));
+    assertEquals(testData, readTable(table, c));
+
+    // Compact the bulk imported file. Subsequent request to load the file should be ignored.
+    c.tableOperations().compact(table, new CompactionConfig().setWait(true));
+
+    Set<Path> tabletFiles = getFiles(c, extent);
+    assertFalse(tabletFiles.contains(bulkLoadPath));
+    assertEquals(1, tabletFiles.size());
+    assertEquals(ImmutableSet.of(bulkLoadPath), getLoaded(c, extent));
+    assertEquals(testData, readTable(table, c));
+
+    // this request should be ignored by the tablet
+    assignMapFiles(fateTxid, asCtx, extent, bulkLoadPath.toString(), status.getLen());
+
+    assertEquals(tabletFiles, getFiles(c, extent));
+    assertEquals(ImmutableSet.of(bulkLoadPath), getLoaded(c, extent));
+    assertEquals(testData, readTable(table, c));
+
+    // this is done to ensure the tablet reads the load flags from the metadata table when it loads
+    c.tableOperations().offline(table, true);
+    c.tableOperations().online(table, true);
+
+    // this request should be ignored by the tablet
+    assignMapFiles(fateTxid, asCtx, extent, bulkLoadPath.toString(), status.getLen());
+
+    assertEquals(tabletFiles, getFiles(c, extent));
+    assertEquals(ImmutableSet.of(bulkLoadPath), getLoaded(c, extent));
+    assertEquals(testData, readTable(table, c));
+
+    // After this, all load request should fail.
+    ZooArbitrator.stop(Constants.BULK_ARBITRATOR_TYPE, fateTxid);
+
+    try {
+      // expect this to fail
+      assignMapFiles(fateTxid, asCtx, extent, bulkLoadPath.toString(), status.getLen());
+      fail();
+    } catch (TApplicationException tae) {
```

Comment:
Could use ExpectedException here.

Reply:
I think that has to be the final step in the test because the exception is caught by junit and therefore jumps out of the function.  I still want to do some verification after this so I don't think I can use that.

Reply:
Interesting.  I thought the ExpectedException worked just like a try/catch, but reported exceptions that didn't get thrown at the end.